src/box/tuple_constraint_def.h

```diff
@@ -185,6 +185,14 @@ tuple_constraint_def_array_dup_raw(const struct tuple_constraint_def *defs,
 				   size_t count, size_t object_size,
 				   size_t additional_size);
 
+/**
+ * Check that all constraints in the `defs` array have unique names.
+ * Return true if the check passed, false otherwise.
+ */
+bool
+tuple_constraint_def_check_names(const struct tuple_constraint_def *defs,
+				 size_t count);
```

Comment:
This function should be named `tuple_constraint_def_array_check` because it checks an array. I'd omit the `_names` suffix for the sake of extensibility. Also, it should return 0 or -1 and set diag that would include the duplicate constraint name. You can use `IllegalParams` error and convert it to the right error code like it's done here:

https://github.com/tarantool/tarantool/blob/78c8cab84197f4a55298d20b7707bd6436c71552/src/box/field_def.c#L349-L350

Reply:
Done.